app/src/main/java/ai/elimu/appstore/synchronization/AppSynchronizationActivity.java

```diff
@@ -17,214 +17,260 @@
 import org.json.JSONException;
 import org.json.JSONObject;
 
+import java.io.IOException;
 import java.lang.reflect.Type;
 import java.util.Calendar;
 import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 import ai.elimu.appstore.BaseApplication;
-import ai.elimu.appstore.BuildConfig;
 import ai.elimu.appstore.R;
 import ai.elimu.appstore.dao.ApplicationDao;
 import ai.elimu.appstore.dao.ApplicationVersionDao;
 import ai.elimu.appstore.model.Application;
 import ai.elimu.appstore.model.ApplicationVersion;
 import ai.elimu.appstore.onboarding.LicenseNumberActivity;
+import ai.elimu.appstore.service.GetAppListByCollectionIdService;
+import ai.elimu.appstore.service.GetAppListService;
 import ai.elimu.appstore.util.ChecksumHelper;
 import ai.elimu.appstore.util.ConnectivityHelper;
 import ai.elimu.appstore.util.DeviceInfoHelper;
-import ai.elimu.appstore.util.JsonLoader;
 import ai.elimu.appstore.util.UserPrefsHelper;
 import ai.elimu.model.enums.admin.ApplicationStatus;
 import ai.elimu.model.gson.admin.ApplicationGson;
 import ai.elimu.model.gson.admin.ApplicationVersionGson;
+import okhttp3.ResponseBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 import timber.log.Timber;
 
 public class AppSynchronizationActivity extends AppCompatActivity {
 
+    private GetAppListService getAppListService;
```

Comment:
This got a little bit messy since it mixes the Retrofit service for both the normal apps and the Custom Project apps. Is there any way to could solve this in a way that would be easier to maintain and understand in the future? Should each URL endpoint have their own activity? Or perhaps we should create a separate productFlavor in build.gradle for the Custom Project variant?